## check ontology helper

In [1]:
import sys
sys.path.append('..')
from ontology_handling import TreeHelper

In [2]:
ontohelper_dharani = TreeHelper(ontoname='dharani')

In [3]:
ontohelper_dharani.print_tree()

[lvl] id (acronym) name
---------------------
 [0] 1 (Br) Brain
   [1] 2 (FB) Forebrain
     [2] 3 (Tel) Telencephalon
       [3] 4 (Ctx) Cerebral cortex
         [*] 5 (SGL-CTX) Subpial granular layer, cerebral cortex
         [*] 6 (MZ-CTX) Marginal zone, cerebral cortex
         [*] 7 (CP-CTX) Cortical plate, cerebral cortex
         [*] 8 (SP-CTX) Subplate zone, cerebral cortex
         [*] 9 (IZ-CTX) Intermediate zone, cerebral cortex
         [*] 10 (SVZ-CTX) Subventricular zone, cerebral cortex
         [*] 11 (VZ-CTX) Ventricular zone, cerebral cortex
         [4] 12 (FCtx) Frontal cortex
           [*] 13 (SGL-FCTx) Subpial granular layer, frontal cortex
           [*] 14 (MZ-FCTx) Marginal zone, frontal cortex
           [*] 15 (CP-FCTx) Cortical plate, frontal cortex
           [*] 16 (SP-FCTx) Subplate zone, frontal cortex
           [*] 17 (IZ-FCTx) Intermediate zone, frontal cortex
           [*] 18 (SVZ-FCTx) Subventricular zone, frontal cortex
           [*] 19 (VZ-FCTx

In [4]:
len(ontohelper_dharani)

496

### search

In [5]:
for ontoid, rec in ontohelper_dharani.onto_lookup.items():
    res = ontohelper_dharani.search(rec.name)
    matched = False
    errors = []
    for elt in res:
        if elt[2]!=ontoid and not matched:
            errors.append(elt)
        else:
            matched = True
        
    if not matched: print('*',end='')
    if len(errors)>0:
        print(ontoid,rec.name,errors)
    

In [6]:
ontohelper_dharani.search('orbital neocortex')

[('orbital cortex', 90.32258064516128, 21)]

In [7]:
ontohelper_dharani.search('ca1',partial=True)

[('subplate zone, ca1 field', 100.0, 420),
 ('intermediate zone, ca1 field', 100.0, 421)]

### subtrees (groups)

In [8]:
ontohelper_dharani.groups

{'HPF': ['HPF', 'HIP'],
 'AMY_BN': ['AMY', 'BN', 'CN'],
 'Mig': ['Lms', 'Rms', 'GE', 'FTS'],
 'HY': ['HY', 'HTH'],
 'TH': ['TH', 'THM'],
 'MB': ['MB', 'M'],
 'HB': ['HB', 'H'],
 'BS': ['BS'],
 'CB': ['CB'],
 'dev': ['dev'],
 'ft': ['ft', 'FWM'],
 'Vs': ['Vs', 'FV'],
 'Ctx': ['Ctx', 'FGM']}

In [9]:
ontohelper_dharani.print_subtree('BS')

[lvl] id (acronym) name
---------------------
 [0] 440 (BS) Brainstem
   [1] 192 (MB) Midbrain
     [2] 203 (Prt) Pretectum
       [3] 322 (APT) Anterior pretectal nucleus
       [3] 323 (MPT) Medial pretectal nucleus
       [3] 324 (PPT) Posterior pretectal nucleus
       [3] 325 (NOT) Nucleus of the optic tract
       [3] 326 (OPT) Olivary pretectal nucleus
     [2] 193 (Tct) Tectum
       [3] 194 (SC) Superior colliculus
       [3] 195 (IC) Inferior colliculus
         [4] 318 (ICc) Inferior colliculus, central nucleus
         [4] 319 (ICe) Inferior colliculus, external nucleus
         [4] 385 (ICd) Inferior colliculus, dorsal nucleus
     [2] 196 (Tgt) Tegmentum
       [3] 320 (PBN) Parabigeminal nucleus
       [3] 321 (NPC) Nucleus of the posterior commissure
       [3] 197 (SN) Substantia nigra
         [4] 198 (SNc) Substantia nigra pars compacta
         [4] 199 (SNr) Substantia nigra pars reticulata
       [3] 200 (R) Red nucleus
       [3] 285 (PAG) Periaqueductal gray
    

In [10]:
ontohelper_dharani.ontoids_by_group['BS']

[440, 192, 204, 210, 448, 445, 446, 447, 460, 461, 462, 463]

many ids are missed - this is because 'HB'(204) is a group by itself, and also a child of 'BS'. Similarly 'MB'(192)

## Special functions

In [11]:
logicalareas = ontohelper_dharani.get_ids_of_cortical_areas()

In [12]:
logicalareas

defaultdict(list,
            {'-FCTx': [13, 14, 15, 16, 17, 18, 19],
             '-ORB': [22, 23, 24, 25, 26, 27, 28],
             '-PAR': [30, 31, 32, 33, 34, 35, 36],
             '-OCC': [39, 40, 41, 42, 43, 44, 45],
             '-TEMP': [48, 49, 50, 51, 52, 53, 54],
             '-ENT': [57, 58, 59, 60, 61, 62, 63],
             '-INS': [500, 501, 502, 503, 504, 505, 506],
             '-CING': [121, 122, 123, 124, 125, 126, 127]})

In [13]:
bylev = ontohelper_dharani.get_ids_by_level(1)

In [14]:
bylev

[2,
 440,
 215,
 317,
 367,
 368,
 369,
 370,
 371,
 158,
 442,
 1023,
 1022,
 1015,
 1021,
 1019,
 1018,
 1016,
 1020,
 1024,
 1000,
 1001,
 1002,
 1003,
 1004,
 1005,
 1006,
 1025,
 1007,
 1008,
 1009,
 1010,
 1013,
 1014,
 1026,
 1027,
 1028,
 455,
 468,
 469,
 466,
 467,
 232,
 235,
 243,
 244,
 415,
 417,
 337,
 338,
 252,
 237,
 238,
 251,
 334,
 414,
 248,
 431,
 343,
 327,
 341,
 342,
 311,
 346,
 347,
 457,
 456,
 250,
 458,
 253,
 225,
 226,
 227,
 228,
 443,
 229,
 230]

In [15]:
ancestorids=ontohelper_dharani.get_ancestor_ids(22)
ancestorids

[1, 2, 3, 4, 21]

In [16]:
ontohelper_dharani.onto_lookup[22]

NodeRecord(acronym='SGL-ORB', name='Subpial granular layer, orbital cortex', level=5, parentid=21, color_hex_triplet='#b8a98a')

In [17]:
nd = ontohelper_dharani._get_node_by_ontoid(21)

In [18]:
ontohelper_dharani._get_node_data(nd)

{'id': 21,
 'name': 'Orbital cortex',
 'acronym': 'ORB',
 'definition (description)': 'Note: may include only the orbital gyrus. Both terms (orbito-frontal ctx and orbital gyrus) appear in coronal and saggital plates, but only the orbital gyrus is listed in the glossary. The orbital sulcus seems to develop at a later g.w.',
 'type (gray matter/fiber tract/CNS cavity/developmental/other)': 'G',
 'parent_structure_id': 4,
 'color_hex_triplet': 'CFE4C2',
 'text': 'Orbital cortex'}

In [19]:
def show_node(node,lev = 0):
    if lev > 1:
        return
    
    for k,v in node.items():
        
        if k!='children':
            print("".join(["\t"]*lev), k, ":", v)
        else:
            for ch in node['children']:
                show_node(ch,lev+1)
        
show_node(nd)

 id : 21
 name : Orbital cortex
 acronym : ORB
 definition (description) : Note: may include only the orbital gyrus. Both terms (orbito-frontal ctx and orbital gyrus) appear in coronal and saggital plates, but only the orbital gyrus is listed in the glossary. The orbital sulcus seems to develop at a later g.w.
 type (gray matter/fiber tract/CNS cavity/developmental/other) : G
 parent_structure_id : 4
 color_hex_triplet : CFE4C2
	 id : 22
	 name : Subpial granular layer, orbital cortex
	 acronym : SGL-ORB
	 definition (description) : Needs a definition
	 type (gray matter/fiber tract/CNS cavity/developmental/other) : D
	 parent_structure_id : 21
	 color_hex_triplet : b8a98a
	 text : Subpial granular layer, orbital cortex
	 id : 23
	 name : Marginal zone, orbital cortex
	 acronym : MZ-ORB
	 definition (description) : See Kostovic et al. 2018
	 type (gray matter/fiber tract/CNS cavity/developmental/other) : D
	 parent_structure_id : 21
	 color_hex_triplet : de895e
	 text : Marginal zone, 

In [20]:
ontohelper_dharani.onto_lookup[230]

NodeRecord(acronym='CC', name='Central canal', level=1, parentid=6000, color_hex_triplet='#231F21')

In [21]:
layeredareas=ontohelper_dharani.get_ids_of_layered_areas()

In [22]:
layeredareas

defaultdict(list,
            {'SGL-': [13,
              22,
              30,
              39,
              48,
              57,
              67,
              75,
              86,
              94,
              102,
              110,
              500,
              121],
             'MZ-': [14,
              23,
              31,
              40,
              49,
              58,
              68,
              76,
              87,
              95,
              103,
              111,
              501,
              122],
             'CP-': [15,
              24,
              32,
              41,
              50,
              59,
              69,
              77,
              88,
              96,
              104,
              112,
              502,
              123],
             'SP-': [16,
              25,
              33,
              42,
              51,
              60,
              70,
              78,
              420,
              422,


In [23]:
ontohelper_dharani.onto_lookup[22]

NodeRecord(acronym='SGL-ORB', name='Subpial granular layer, orbital cortex', level=5, parentid=21, color_hex_triplet='#b8a98a')

In [24]:
ontohelper_dharani.onto_lookup[5]

NodeRecord(acronym='SGL-CTX', name='Subpial granular layer, cerebral cortex', level=4, parentid=4, color_hex_triplet='#b8a98a')

In [25]:
ontohelper_dharani.get_group_by_acronym('ORB')

'Ctx'

In [26]:
ontohelper_dharani.ontoids_by_group['MB']

[185,
 407,
 408,
 192,
 203,
 322,
 323,
 324,
 325,
 326,
 193,
 194,
 195,
 318,
 319,
 385,
 196,
 320,
 321,
 197,
 198,
 199,
 200,
 285,
 444,
 403,
 404,
 263,
 262,
 261,
 201,
 428,
 449,
 283,
 450,
 451,
 452,
 454,
 453,
 377]

In [27]:
ontohelper_dharani.onto_lookup[185]

NodeRecord(acronym='M', name='Mammillary body', level=4, parentid=177, color_hex_triplet='#B48EBC')

In [28]:
ungroupedids = []
for id,rec in ontohelper_dharani.onto_lookup.items():
    if ontohelper_dharani.get_group_by_acronym(rec.acronym) is None:
        ungroupedids.append(id)

In [29]:
[ontohelper_dharani.onto_lookup[id] for id in ungroupedids]

[NodeRecord(acronym='Br', name='Brain', level=0, parentid=0, color_hex_triplet='#F6FAF4'),
 NodeRecord(acronym='FB', name='Forebrain', level=1, parentid=1, color_hex_triplet='#EFF6EA'),
 NodeRecord(acronym='Tel', name='Telencephalon', level=2, parentid=2, color_hex_triplet='#EFF6EA'),
 NodeRecord(acronym='Di', name='Diencephalon', level=2, parentid=2, color_hex_triplet='#FFF3F3'),
 NodeRecord(acronym='SP', name='Spinal cord', level=0, parentid=0, color_hex_triplet='#C0C1C2'),
 NodeRecord(acronym='DGSC', name='Dorsal gray of the spinal cord', level=1, parentid=231, color_hex_triplet='#AFB0B1'),
 NodeRecord(acronym='DH', name='Dorsal horn', level=1, parentid=231, color_hex_triplet='#AFB0B1'),
 NodeRecord(acronym='IH', name='Intermediate gray of the spinal cord', level=1, parentid=231, color_hex_triplet='#B1B2B3'),
 NodeRecord(acronym='CG', name='Central gray of the spinal cord', level=1, parentid=231, color_hex_triplet='#ACADAE'),
 NodeRecord(acronym='VG', name='Ventral gray of the spina

## Searchable tree display (jstree)

Shows the Dharani ontology as an embedded jstree (https://www.jstree.com/) with a search box

In [30]:
from IPython.display import display, HTML
import json

In [31]:
tree_data = json.dumps(ontohelper_dharani.treenom)

In [32]:
if 'html_head' not in locals():
    html_head="""
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jstree/3.3.12/themes/default/style.min.css" />
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jstree/3.3.12/jstree.min.js"></script>
    """

    display(HTML(html_head))

In [33]:
html_code = """
    <input type="text" id="search_input" placeholder="Search tree..." style="margin-bottom:10px;width:200px;padding:5px;"/>
    <div id="jstree_demo"></div>

    <script>
        $(document).ready(function() {
            // Initialize jstree with search plugin
            $('#jstree_demo').jstree({
                
                'core': {
                    'multiple' : false,
                    'animation' : 0,
                    'themes': {'icons':false,},
                    'data': %s,
                },                
                'plugins': ['search'],
                'search': {
                    'show_only_matches':true,
                    'show_only_matches_children':true,
                }
            });
            
            // Bind search input to jstree search function
            $('#search_input').on('keyup', function() {
                var searchValue = $(this).val();
                $('#jstree_demo').jstree(true).search(searchValue);
            });
        });
    </script>
"""

Note: if the below cell does not show two elements (brain, spinal cord), restart the notebook and retry

In [34]:
display(HTML(html_code % tree_data))

## Fuzzy match into Allen ontology

In [27]:
ontohelper_allen = TreeHelper('allen_devhuman')

In [28]:
len(ontohelper_allen)

3315

In [29]:

def nameupdate(name):
    expansions = {
        'SG':'subpial granular layer',
        'MZ': 'marginal zone',
        'CP': 'cortical plate',
        'SP': 'subplate zone',
        'IZ': 'intermediate zone',
        'SZ': 'subventricular zone',
        'VZ': 'ventricular zone',
        'Hipp': 'hippocampus',
        # 'GE': 'ganglionic eminence',
    }
    if ' in ' in name:
        for acr,exp in expansions.items():
            name = name.replace(acr,exp)
        name = name.replace(' in ', ', ')
        
    if 'neocortex' in name:
        name = name.replace('neocortex','cortex')
    return name

updated_allen_dict = {k:nameupdate(v.name) for k,v in ontohelper_allen.onto_lookup.items()}

ontohelper_allen.search_dict = updated_allen_dict


In [30]:
ontohelper_allen.search('subiculum')

[('subiculum', 100.0, 10302)]

In [31]:
from collections import defaultdict
from tqdm import tqdm

matches = defaultdict(list)

for dh_id,rec in tqdm(ontohelper_dharani.onto_lookup.items()):
    query = rec.name.lower()# .replace(',', ' in')
    ret = ontohelper_allen.search(query)
    
    matches[dh_id].extend(ret)

    if len(matches[dh_id])>1:
        print(query,matches[dh_id])

 13%|█▎        | 66/496 [00:00<00:00, 605.51it/s]

forebrain [('forebrain (prosencephalon)', 100.0, 10156), ('basal forebrain', 100.0, 10349)]
subpial granular layer, cerebral cortex [('subpial granular layer, perirhinal cortex', 90.0, 13213), ('subpial granular layer, parietal cortex', 89.74358974358975, 11593), ('subpial granular layer, temporal cortex', 89.74358974358975, 11598), ('subpial granular layer, frontal cortex', 88.31168831168831, 11581), ('subpial granular layer, periallocortex', 88.31168831168831, 12974)]
marginal zone, cerebral cortex [('marginal zone, perirhinal cortex', 87.09677419354838, 11645), ('marginal zone, parietal cortex', 86.66666666666667, 10511), ('marginal zone, temporal cortex', 86.66666666666667, 10512)]
cortical plate, cerebral cortex [('cortical plate, perirhinal cortex', 87.5, 11746), ('cortical plate, parietal cortex', 87.09677419354838, 10518), ('cortical plate, temporal cortex', 87.09677419354838, 10519), ('cortical plate, frontal cortex', 85.24590163934427, 10516), ('cortical plate, periallocortex

 26%|██▌       | 127/496 [00:01<00:03, 106.22it/s]

subventricular zone, parasubiculum [('subventricular zone', 100.0, 10536), ('ventricular zone', 100.0, 10542), ('subiculum', 100.0, 10302)]
ventricular zone, parasubiculum [('ventricular zone', 100.0, 10542), ('subiculum', 100.0, 10302)]
cortical plate, postsubiculum [('cortical plate', 100.0, 10515), ('subiculum', 100.0, 10302)]
subplate zone, postsubiculum [('subplate zone', 100.0, 10522), ('subiculum', 100.0, 10302)]
subventricular zone, postsubiculum [('subventricular zone', 100.0, 10536), ('ventricular zone', 100.0, 10542), ('subiculum', 100.0, 10302)]
ventricular zone, postsubiculum [('ventricular zone', 100.0, 10542), ('subiculum', 100.0, 10302)]
marginal zone, insula [('marginal zone', 100.0, 10508), ('marginal zone, insular cortex', 95.23809523809523, 11634), ('marginal zone, dysgranular insular cortex', 95.23809523809523, 11635), ('marginal zone, granular insular cortex', 95.23809523809523, 11636), ('marginal zone, agranular insular cortex', 95.23809523809523, 13279)]
cortica

 31%|███▏      | 156/496 [00:01<00:03, 111.01it/s]

olfactory cortex [('subpial granular layer, olfactory entorhinal cortex', 100.0, 12976), ('marginal zone, olfactory entorhinal cortex', 100.0, 11642), ('cortical plate, olfactory entorhinal cortex', 100.0, 11743)]
amygdala [('extended amygdala', 100.0, 10383), ('sublenticular extended amygdala', 100.0, 10387)]
basomedial nucleus of the amygdala [('medial nucleus', 100.0, 10376), ('median nucleus', 92.85714285714286, 12411)]
basomedial nucleus of the amygdala, parvocellular part [('medial nucleus', 100.0, 10376), ('median nucleus', 92.85714285714286, 12411)]
lateral nucleus of the amygdala [('lateral nucleus', 100.0, 10367), ('lateral nucleus', 100.0, 12410)]


 47%|████▋     | 234/496 [00:01<00:01, 138.77it/s]

dorsal complex of the thalamus [('thalamus', 100.0, 10390), ('dorsal thalamus', 100.0, 10391)]
nucleus fasciculosus of the thalamus [('thalamus', 100.0, 10390), ('fasciculosus nucleus of thalamus', 100.0, 13047)]
lateral dorsal nucleus of the thalamus [('lateral dorsal nucleus of thalamus', 94.44444444444444, 10396), ('central dorsal nucleus of thalamus', 86.11111111111111, 13331), ('anterodorsal nucleus of thalamus', 85.71428571428572, 10393)]
lateral posterior nucleus of the thalamus [('lateral posterior nucleus of thalamus', 94.87179487179486, 10408), ('ventral posterior nucleus of thalamus', 87.17948717948718, 10423)]
periventricular complex of the thalamus [('thalamus', 100.0, 10390), ('periventricular area of thalamus', 91.52542372881356, 13046)]
parafascicular nucleus of the thalamus [('parafascicular nucleus of thalamus', 94.44444444444444, 10450), ('subparafascicular nucleus of thalamus', 90.66666666666666, 13054)]
paraventricular nucleus of the thalamus [('paraventricular nuc

 55%|█████▌    | 275/496 [00:02<00:01, 137.05it/s]

supraoptic nucleus of the hypothalamus [('hypothalamus', 100.0, 10467), ('supraoptic nucleus', 100.0, 10481)]
suprachiasmatic nucleus of the hypothalamus [('hypothalamus', 100.0, 10467), ('suprachiasmatic nucleus', 100.0, 10480)]
retrochiasmatic nucleus of the hypothalamus [('hypothalamus', 100.0, 10467), ('retrochiasmatic nucleus', 100.0, 15536)]
dorsomedial nucleus of the hypothalamus [('hypothalamus', 100.0, 10467), ('dorsomedial nucleus', 100.0, 12409), ('dorsomedial nucleus of Pr5', 91.66666666666667, 12445)]
lateral hypothalamic area [('anterior hypothalamic area', 86.27450980392157, 266441551), ('dorsal hypothalamic area', 85.71428571428572, 13335)]
ventral premammillary nucleus [('premammillary nulceus', 95.23809523809523, 13072), ('supramammillary nucleus', 90.9090909090909, 10497), ('red nucleus', 90.9090909090909, 12247)]
preoptic area [('medial preoptic area', 100.0, 266441547), ('lateral preoptic area', 100.0, 10471)]
preoptic periventricular nucleus [('paraventricular nuc

 62%|██████▎   | 310/496 [00:02<00:01, 134.50it/s]

inferior colliculus, external nucleus [('nucleus U', 94.11764705882352, 12242), ('nucleus K', 94.11764705882352, 12522), ('nucleus L', 94.11764705882352, 12523), ('nucleus Y', 94.11764705882352, 12652)]
tegmentum [('midbrain tegmentum', 100.0, 12195), ('other nuclei, midbrain tegmentum', 100.0, 12277), ('paracollicular tegmentum', 100.0, 12282), ('pontine tegmentum', 100.0, 12416)]
nucleus of the posterior commissure [('magnocellular nucleus of the posterior commissure', 100.0, 12279), ('posterior commissure', 100.0, 10567), ('interstitial nucleus of posterior limb of anterior commissure', 93.93939393939394, 10388), ('interstitial nucleus of posterior commissure', 93.93939393939394, 12194)]
substantia nigra pars compacta [('substantia nigra', 100.0, 12251), ('substantia nigra, compact part', 93.33333333333333, 12252)]
substantia nigra pars reticulata [('substantia nigra', 100.0, 12251), ('substantia nigra, reticular part', 90.625, 12259)]
periaqueductal gray [('periaqueductal gray subs

 70%|██████▉   | 345/496 [00:02<00:01, 134.24it/s]

nucleus of the lateral lemniscus, dorsal [('lateral nucleus', 100.0, 10367), ('dorsal nucleus', 100.0, 12407), ('lateral nucleus', 100.0, 12410), ('dorsal nucleus of lateral lemniscus', 93.33333333333333, 12455)]
nucleus of the lateral lemniscus, ventral [('lateral nucleus', 100.0, 10367), ('lateral nucleus', 100.0, 12410), ('ventral nucleus of lateral lemniscus', 93.50649350649351, 12457)]
parabrachial nucleus [('subparabrachial nucleus', 93.02325581395348, 12491), ('parabrachial nuclei', 92.3076923076923, 12481), ('medial parabrachial nucleus', 85.1063829787234, 12487), ('pararaphales nucleus', 85.0, 12655)]
parabrachial nucleus, lateral part [('lateral parabrachial nucleus, central part', 100.0, 12483), ('lateral parabrachial nucleus, dorsal part', 100.0, 12484), ('lateral parabrachial nucleus, external part', 100.0, 12485), ('lateral parabrachial nucleus, superior part', 100.0, 12486), ('lateral parabrachial nucleus', 90.3225806451613, 12482)]
reticular tegmental nucleus [('reticul

 78%|███████▊  | 386/496 [00:03<00:00, 113.99it/s]

anterior lobe of cerebellum (h_v) [('cerebellum', 100.0, 10656), ('anterior lobe', 100.0, 12838)]
cerebellar nuclei [('precerebellar nuclei', 91.89189189189189, 12595), ('cerebellar deep nuclei', 87.17948717948718, 10660)]
interposed nucleus [('nucleus U', 94.11764705882352, 12242), ('nucleus K', 94.11764705882352, 12522), ('nucleus L', 94.11764705882352, 12523), ('nucleus Y', 94.11764705882352, 12652)]


 83%|████████▎ | 413/496 [00:03<00:00, 97.24it/s] 

ganglionic eminence [('lateral ganglionic eminence', 100.0, 10551), ('medial ganglionic eminence', 100.0, 10550)]
migratory streams [('migratory streams, hindbrain', 100.0, 12722), ('caudal (posterior) extramural migratory stream', 96.96969696969697, 12724), ('intramural migratory stream', 96.96969696969697, 12725)]
parietal lobe neuroepithelium and subventricular zone [('subventricular zone', 100.0, 10536), ('parietal lobe', 100.0, 12131)]
occipital lobe neuroepithelium and subventricular zone [('subventricular zone', 100.0, 10536), ('occipital lobe', 100.0, 12148)]


 91%|█████████▏| 453/496 [00:03<00:00, 106.25it/s]

fiber tracts [('forebrain commissural fiber tracts', 100.0, 10558), ('forebrain ipsilateral fiber tracts', 100.0, 10568)]
external medullary lamina of the thalamus [('external medullary lamina of thalamus', 94.87179487179486, 12038), ('internal medullary lamina of thalamus', 89.74358974358975, 12063)]
internal medullary lamina of the thalamus [('internal medullary lamina of thalamus', 94.87179487179486, 12063), ('external medullary lamina of thalamus', 89.74358974358975, 12038)]


100%|██████████| 496/496 [00:04<00:00, 123.83it/s]


In [32]:
for k,v in matches.items():
    name = ontohelper_dharani.onto_lookup[k].name
    if len(v)==1:
        pass
        # print('*',k,name,v)
    else:
        print(k,name,'---', v)
    

2 Forebrain --- [('forebrain (prosencephalon)', 100.0, 10156), ('basal forebrain', 100.0, 10349)]
5 Subpial granular layer, cerebral cortex --- [('subpial granular layer, perirhinal cortex', 90.0, 13213), ('subpial granular layer, parietal cortex', 89.74358974358975, 11593), ('subpial granular layer, temporal cortex', 89.74358974358975, 11598), ('subpial granular layer, frontal cortex', 88.31168831168831, 11581), ('subpial granular layer, periallocortex', 88.31168831168831, 12974)]
6 Marginal zone, cerebral cortex --- [('marginal zone, perirhinal cortex', 87.09677419354838, 11645), ('marginal zone, parietal cortex', 86.66666666666667, 10511), ('marginal zone, temporal cortex', 86.66666666666667, 10512)]
7 Cortical plate, cerebral cortex --- [('cortical plate, perirhinal cortex', 87.5, 11746), ('cortical plate, parietal cortex', 87.09677419354838, 10518), ('cortical plate, temporal cortex', 87.09677419354838, 10519), ('cortical plate, frontal cortex', 85.24590163934427, 10516), ('cortic

In [41]:
ontohelper_allen.search('pontine',partial=True)

[('prefrontopontine fibers', 100.0, 12042),
 ('frontopontine fibers', 100.0, 12048),
 ('parietopontine fibers', 100.0, 12055),
 ('occipitopontine fibers', 100.0, 12056),
 ('temporopontine fibers', 100.0, 12061)]

In [42]:
for ontoid, rec in ontohelper_allen.onto_lookup.items():
    if 'field' in rec.name.lower():
        print(ontoid, rec)

10462 NodeRecord(acronym='FF', name='nucleus of the field of Forel', level=6, parentid=10461, color_hex_triplet='#B67BB4')
15550 NodeRecord(acronym='CTF', name='central tegmental field', level=5, parentid=12277, color_hex_triplet='#6CA9BF')
12289 NodeRecord(acronym='RRF', name='retrorubral field', level=5, parentid=12277, color_hex_triplet='#5298AE')


In [43]:
ontohelper_allen.onto_lookup[11737]

NodeRecord(acronym='CPhipc', name='CP in caudal hippocampus', level=7, parentid=11735, color_hex_triplet='#5D457F')

In [44]:
ontohelper_allen.onto_lookup[11735]

NodeRecord(acronym='CPhip', name='CP in hippocampal proper', level=6, parentid=10520, color_hex_triplet='#5D457F')

In [45]:
ontohelper_dharani.onto_lookup[450]

NodeRecord(acronym='IPNdl', name='Interpeduncular nucleu, dorsolateral part', level=5, parentid=283, color_hex_triplet='#8EBDCD')